### PR TITLE
Fix Sequential RS linting

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #915
* #879
* __->__ #982

Differential Revision: [D42979396](https://our.internmc.facebook.com/intern/diff/D42979396)

### CODE REVIEW

File: torchdata/dataloader2/utils/worker.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def process_reset_fn(
     datapipe: DataPipe,
     worker_info: WorkerInfo,
     seed_generator: SeedGenerator,
-    custom_reset_fn: Optional[Callable[[DataPipe, WorkerInfo], DataPipe]] = None,
+    custom_reset_fn: Optional[Callable[[DataPipe, WorkerInfo, SeedGenerator], DataPipe]] = None,
```
Comment (NivekT) [inline torchdata/dataloader2/utils/worker.py:145]: Not sure why that was removed in the last PR

File: torchdata/dataloader2/reading_service.py [hunk 2]
```diff
@@ -537,12 +547,14 @@ def finalize_iteration(self) -> None:
...
     # Sequential Order
     def checkpoint(self) -> bytes:
-        states = []
-        for rs in self.reading_services:
-            states.append(rs.checkpoint())
+        # states = []
+        # for rs in self.reading_services:
+        #     states.append(rs.checkpoint())
+        # return states
+        raise NotImplementedError
```
Comment (NivekT) [inline torchdata/dataloader2/reading_service.py:554]: The implementation doesn't do anything (yet). Let's raise an exception instead?

File: torchdata/dataloader2/reading_service.py [hunk 3]
```diff
@@ -485,7 +491,7 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
         self._datapipe = datapipe
         return datapipe
 
-    def initialize_iteration(self, seed_generator: SeedGenerator) -> None:
+    def initialize_iteration(self, seed_generator: SeedGenerator, _=None) -> None:
```
Comment (NivekT) [inline torchdata/dataloader2/reading_service.py:494]: The superclass signature wants a `iter_reset_fn` as the 2nd argument but I don't think it is relevant here.

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42979396).

**Review by ejguan — COMMENTED** [review]
Emmm. I actually have a PR in https://github.com/pytorch/data/pull/983 that covers the lint fix. Do you mind using mine since it covers more problems.

**Pull request closed without merging**